app/src/components/search/content.tsx

```diff
@@ -29,44 +42,71 @@ const SearchContent = () => {
       <h2>Search params: {query} </h2>
       <br />
       <Box sx={{ display: "flex", gap: "xs", marginBottom: "l" }}>
-        <Text sx={{ fontWeight: "500", marginBottom: 0, marginTop: "xs" }}>
-          {" "}
-          Sort by{" "}
-        </Text>{" "}
-        <Menu
-          //showSelectionAsLabel
-          showLabel
-          selectedItem="chronological-descending"
-          labelText={"Sort By"}
-          listItemsData={[
-            {
-              id: "chronological-descending",
-              label: "Newest to oldest",
-              onClick: onMenuClick,
-              type: "action",
-            },
-            {
-              id: "chronological-ascending",
-              label: "Oldest to newest",
-              onClick: onMenuClick,
-              type: "action",
-            },
-            {
-              id: "alphabetical-descending",
-              label: "Title A to Z",
-              onClick: onMenuClick,
-              type: "action",
-            },
-            {
-              id: "alphabetical-ascending",
-              label: "Title Z to A",
-              onClick: onMenuClick,
-              type: "action",
-            },
-          ]}
-        />
+        {showFilter ? (
+          <>
+            <Text sx={{ fontWeight: "500", marginBottom: 0, marginTop: "xs" }}>
+              {" "}
+              Sort by{" "}
+            </Text>{" "}
+            <Menu
+              //showSelectionAsLabel
+              showLabel
+              selectedItem="chronological-descending"
+              labelText={"Sort By"}
+              listItemsData={[
+                {
+                  id: "chronological-descending",
+                  label: "Newest to oldest",
+                  onClick: onMenuClick,
+                  type: "action",
+                },
+                {
+                  id: "chronological-ascending",
+                  label: "Oldest to newest",
+                  onClick: onMenuClick,
+                  type: "action",
+                },
+                {
+                  id: "alphabetical-descending",
+                  label: "Title A to Z",
+                  onClick: onMenuClick,
+                  type: "action",
+                },
+                {
+                  id: "alphabetical-ascending",
+                  label: "Title Z to A",
+                  onClick: onMenuClick,
+                  type: "action",
+                },
+              ]}
+            />
+          </>
+        ) : (
+          <></>
+        )}
       </Box>
-      <CollectionsGrid data={mockCollections} />
+
+      {/* certainly a better way to handle this but it's annoying me. */}
```

Comment:
I agree there must be a better way. The way it is now, the component is coupled to the URL pathname of where it's rendered. It should not care what the pathname is.

This might be a case where reusability is not helpful. I don't want to suggest prop drilling but you can try doing so from the parent page component that set a flag such as `isSearchPage` instead of using the `pathname` comparison. Let me know what you think about this.

Reply:
I like the `isSearchPage` approach